api/v1beta1/glanceapi_webhook.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+// GlanceAPIDefaults -
+type GlanceAPIDefaults struct {
+	ContainerImageURL string
+}
+
+var glanceAPIDefaults GlanceAPIDefaults
+
+// log is for logging in this package.
+var glanceapilog = logf.Log.WithName("glanceapi-resource")
+
+// SetupGlanceAPIDefaults - initialize GlanceAPI spec defaults for use with either internal or external webhooks
+func SetupGlanceAPIDefaults(defaults GlanceAPIDefaults) {
+	glanceAPIDefaults = defaults
+	glancelog.Info("Glance defaults initialized", "defaults", defaults)
+}
```

Comment:
I think we also need something that pulls-in the ENV vars and sets them in the `GlanceAPIDefaults`, like we do here [1].  Maybe I just am missing where you're already doing this, however. 

[1] https://github.com/openstack-k8s-operators/glance-operator/blob/66fbd23a85871d6c734bd49d0af628e4b50ec23b/api/v1beta1/common.go#L114-L121

Reply:
You're right, I need that function too in `common.go`. Updating the patch in a moment.

Reply:
Done